packages/db/migrations/seed.ts

```diff
@@ -48,21 +52,82 @@ const seedOnboardingAsync = async (db: Database) => {
   console.log("Created onboarding step through seed");
 };
 
+const seedDefaultSearchEnginesAsync = async (db: Database) => {
+  const existingSearchEngines = await db.$count(searchEngines);
+
+  if (existingSearchEngines > 0) {
+    console.log("Skipping seeding of default search engines as some already exists");
+    return;
+  }
+
+  const homarrId = createId();
+  const defaultSearchEngines = [
+    {
+      id: createId(),
+      name: "Google",
+      iconUrl: "https://cdn.jsdelivr.net/gh/homarr-labs/dashboard-icons/svg/google.svg",
+      short: "g",
+      description: "Search the web with Google",
+      urlTemplate: "https://www.google.com/search?q=%s",
+      type: "generic" as const,
+      integrationId: null,
+    },
+    {
+      id: createId(),
+      name: "YouTube",
+      iconUrl: "https://cdn.jsdelivr.net/gh/homarr-labs/dashboard-icons/svg/youtube.svg",
+      short: "yt",
+      description: "Search for videos on YouTube",
+      urlTemplate: "https://www.youtube.com/results?search_query=%s",
+      type: "generic" as const,
+      integrationId: null,
+    },
+    {
+      id: homarrId,
+      name: "Homarr Docs",
+      iconUrl: "https://cdn.jsdelivr.net/gh/homarr-labs/dashboard-icons/svg/homarr.svg",
+      short: "docs",
+      description: "Search the Homarr documentation",
+      urlTemplate: "https://homarr.dev/search?q=%s",
+      type: "generic" as const,
+      integrationId: null,
+    },
+  ];
+
+  await db.insert(searchEngines).values(defaultSearchEngines);
+  console.log(`Created ${defaultSearchEngines.length} default search engines through seeding process`);
+
+  // Set Homarr docs as the default search engine in server settings
+  const searchSettings = await getServerSettingByKeyAsync(db, "search");
+
+  if (searchSettings) {
+    if (!searchSettings.defaultSearchEngineId) {
+      await updateServerSettingByKeyAsync(db, "search", {
+        ...searchSettings,
+        defaultSearchEngineId: homarrId,
+      });
+      console.log("Set Homarr docs as the default search engine");
+    }
+  } else {
+    await updateServerSettingByKeyAsync(db, "search", {
+      defaultSearchEngineId: homarrId,
+    });
+    console.log("Created search settings with Homarr docs as the default search engine");
+  }
```

Comment:
Remove this as the server-settings are always initialized before it and so this is not necessary (and would be wrong if we would like to create it)

Reply:
What I mean is, that the seeding of the server-settings happens first already, so we will never get into this else clause.
AND
if we would have needed this else clause, then the server settings should not be updated and rather inserted

Reply:
Thanks ! Much clearer now :D I've committed d289fe33c2cba3cabe092b10d8f07ee9e09be461 to address this